dataflow/lib/exporters.py

```diff
@@ -0,0 +1,80 @@
+from functools import wraps
+
+def text(datasets, export_method="to_column_text", headers=None, concatenate=False):
+    import json
+    header_string = "#" + json.dumps(headers, separators=(',', ':'))[1:-1] + "\n"
+    outputs = []
+    if len(datasets) > 0:
+        exports = [getattr(d, export_method)() for d in datasets]
+        if concatenate:
+            first = exports[0]
+            name = first.get("name", "default_name")
+            entry = first.get("entry", "entry")
+            file_suffix = first.get("file_suffix", "dat")
+            filename = "%s_%s.%s" % (name, entry, file_suffix)
+            export_strings = [e['value'] for e in exports]
+            export_string = header_string + "\n\n".join(export_strings)
+            outputs.append({"filename": filename, "value": export_string})
+        else:
+            for i, e in enumerate(exports):
+                export_string = header_string + e["value"]
+                name = e.get("name", "default_name_%d" %(i,))
+                entry = e.get("entry", "entry")
+                file_suffix = e.get("file_suffix", "dat")
+                filename = "%s_%s.%s" % (name, entry, file_suffix)
+                outputs.append({"filename": filename, "value": export_string})
+    return outputs
+
+def hdf(datasets, export_method="to_hdf", headers=None, concatenate=False):
+    import h5py
+    import json
+    import io
+    header_string = json.dumps(headers, separators=(',', ':'))
+    outputs = []
+    if len(datasets) > 0:
+        exports = [getattr(d, export_method)() for d in datasets]
+        if concatenate:
+            first = exports[0]
+            name = first.get("name", "default_name")
+            entry = first.get("entry", "entry")
+            file_suffix = first.get("file_suffix", "dat")
+            filename = "%s_%s.%s" % (name, entry, file_suffix)
+            fid = io.BytesIO()
+            container = h5py.File(fid)
+            container.attrs["template_def"] = header_string
```

Comment:
You aren't defining the NXroot metadata, but I'm not sure any readers will care.

Here's what I do in nice (h5nexus.py):
```
        attrs = {
            'NX_class': 'NXroot',
            'file_name': filename,
            'file_time': timestr,
            'HDF5_Version': h5version,
            'NeXus_version': NEXUS_VERSION,
        }
```

Reply:
Sure, we can do that.  I was aware that the nexus file output is not fully compliant, so if you want to make it more so that's fine.